app.py

```diff
@@ -0,0 +1,172 @@
+    """
+    # Get the ids of the Slack user, channel, reaction, and timestamp of message reacted to associated with the incoming event
+    channel_id = event.get("item", {}).get("channel")
+    user_id = event.get("user")
+    reaction = event.get("reaction")
+    message_ts = event.get("item", {}).get("ts")
+    message_thread_ts = event.get("item", {}).get("thread_ts")
+
+    # reactions and message only logged if the reaction is a round pushpin
+    if reaction == "round_pushpin":
+        # initializes the file for the messages
+        file = open("PythOnBoardingBot/messages.txt", "r+")
+        file.seek(len(file.read(-1)))
+
+        try:
+            # Call the conversations.history method using the WebClient
+            # The client passes the token you included in initialization    
+            result = client.conversations_history(
+                channel=channel_id,
+                inclusive=True,
+                oldest=message_ts,
+                latest=message_ts,
+                limit=1
+            )
+            message = result["messages"][0]
+
+            # Print message text
+            logger.info("{} has reacted with a '{}' to {}".format(user_id, reaction, message["text"]))
+            client.chat_postMessage(channel=channel_id, text="What do you want to title this wiki?", thread_ts=message_ts)
+
+            #  writes the message that was being reacted to into the messages.txt file
+            file.write("{}\n".format(message["text"]))
+            file.close()
+
+        except SlackApiError as e:
+            logger.info(f"Error: {e}")
+
+
+
+# =============== Pin Added Events ================ #
+# When a users pins a message the type of the event will be 'pin_added'.
+# Here we'll link the update_pin callback to the 'pin_added' event.
+@app.event("pin_added")
+def update_pin(event, client):
+    """Update the onboarding welcome message after receiving a "pin_added"
+    event from Slack. Update timestamp for welcome message as well.
+    """
+    # Get the ids of the Slack user and channel associated with the incoming event
+    channel_id = event.get("channel_id")
+    user_id = event.get("user")
+
+    # Get the original tutorial sent.
+    onboarding_tutorial = onboarding_tutorials_sent[channel_id][user_id]
+
+    # Mark the pin task as completed.
+    onboarding_tutorial.pin_task_completed = True
+
+    # Get the new message payload
+    message = onboarding_tutorial.get_message_payload()
+
+    # Post the updated message in Slack
+    updated_message = client.chat_update(**message)
+
+
+# ============== Message Events ============= #
+# When a user sends a message to a channel, the event type will be 'message'.
+# Here we'll link the message callback to the 'message' event.
+@app.event("message")
+def message(event, client):
+    # initializes the file for the messages and puts cursor at the last line
+    file = open("PythOnBoardingBot/messages.txt", "r+")
+    file.seek(len(file.read(-1)))

```

Comment:
You open the file later in this function, looks like you only need to keep one of them

Reply:
Oops- small mistake while copying & pasting code. Lines 137-139 have been deleted.